### PR TITLE
4 get api articles article

### PR DESCRIPTION
**Pull request by ZEwela**
returns an article object specified by article_id

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -84,6 +84,47 @@ describe("GET /api", () => {
...
+      votes: 100,
+      article_img_url:
+        "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+    };
+    return request(app)
+      .get("/api/articles/1")
+      .expect(200)
+      .then((response) => {
+        const article = response.body.article;
+        expect(article).toEqual(expectedOutput);
```
Comment (nc-saleh) [inline __tests__/app.test.js:105]: We need to be careful when we are asserting the structure of an object like this. What if we were to decide that we wanted to add a property to our data? We would have to amend this test for every property that we wanted to add if we are going along the line of checking the complete shape of the object. 

We can however target properties that we are expecting to be on that object and specifically assert that they will have a specific value.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -84,6 +84,47 @@ describe("GET /api", () => {
...
+      });
+  });
+  test("STATUS 400: returns an error when passed non-existent but valid article_id", () => {
+    return request(app)
+      .get("/api/articles/9999")
+      .expect(400)
```
Comment (nc-saleh) [inline __tests__/app.test.js:111]: this should be a 404 - it's perfectly reasonable to assume there could be 9999 articles on a very popular forum/social media page.

File: models/articles.model.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+const db = require("../db/connection");
+
+function selectArticleById(articleId) {
+  return db
+    .query(`SELECT * FROM articles WHERE article_id = $1`, [articleId])
+    .then((response) => {
+      const article = response.rows[0];
+      if (!article) {
+        return Promise.reject({ status: 400, msg: "Bad request." });
```
Comment (nc-saleh) [inline models/articles.model.js:9]: 404 - not found

File: __tests__/app.test.js (hunk 1 above)
Comment (ZEwela) [reply __tests__/app.test.js:105]: done! thank you!

**Pull request merged**